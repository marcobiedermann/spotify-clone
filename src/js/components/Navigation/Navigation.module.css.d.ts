declare const styles: {
  readonly navigation__list: string;
  readonly navigation__link: string;
};
export = styles;
