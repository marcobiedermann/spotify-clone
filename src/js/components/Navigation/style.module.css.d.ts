declare const styles: {
  readonly navigation__list: string;
};
export = styles;
