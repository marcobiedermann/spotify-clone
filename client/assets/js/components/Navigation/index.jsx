import React from 'react';
import { Link } from 'react-router-dom';
import * as routes from '../../constants/routes';

const Navigation = () => (
  <nav>
    <ul>
      <li>
        <Link to={routes.INDEX}>
          Home
        </Link>
      </li>
      <li>
        <Link to={routes.BROWSE}>
          Browse
        </Link>
      </li>
    </ul>
  </nav>
);

export default Navigation;
