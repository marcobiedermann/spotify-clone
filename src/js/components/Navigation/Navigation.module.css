.navigation__list {
  list-style: none;
  margin: 0;
  padding: 0;
}

.navigation__link {
  display: inline-block;
  font-size: 0.875rem;
  padding: 0.35em 1.75em;
}

.navigation__link:focus,
.navigation__link:hover {
  text-decoration: none;
}
