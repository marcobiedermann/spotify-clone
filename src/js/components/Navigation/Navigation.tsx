import React from 'react';
import { Link } from 'react-router-dom';

interface Item {
  id: string;
  name: string;
  path: string;
  items?: Item[];
}

interface NavigationListProps {
  items: Item[];
}

function NavigationList(props: NavigationListProps): JSX.Element {
  const { items } = props;

  return (
    <ul>
      {items.map((item) => {
        const { id, name, path } = item;

        return (
          <li key={id}>
            <Link to={path}>{name}</Link>
            {item.items && <NavigationList items={item.items} />}
          </li>
        );
      })}
    </ul>
  );
}

interface NavigationProps {
  items: Item[];
}

function Navigation(props: NavigationProps): JSX.Element {
  const { items } = props;

  return (
    <nav>
      <NavigationList items={items} />
    </nav>
  );
}

export type { NavigationProps };
export default Navigation;
