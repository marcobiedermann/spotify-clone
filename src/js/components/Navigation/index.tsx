import React from 'react';
import { Link } from 'react-router-dom';
import { navigation__list } from './style.module.css';

function Navigation(): JSX.Element {
  return (
    <nav>
      <ul className={navigation__list}>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/browse">Browse</Link>
        </li>
      </ul>
    </nav>
  );
}

export default Navigation;
