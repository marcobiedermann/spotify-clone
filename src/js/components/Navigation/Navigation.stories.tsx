import type { Meta, StoryObj } from '@storybook/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '.';

const meta: Meta<typeof Navigation> = {
  component: Navigation,
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
};

type Story = StoryObj<typeof Navigation>;

const Default: Story = {};

export { Default };
export default meta;
