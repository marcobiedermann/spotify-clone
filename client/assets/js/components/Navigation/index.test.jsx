import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '.';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
