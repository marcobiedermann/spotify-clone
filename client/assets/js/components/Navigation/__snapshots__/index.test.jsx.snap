// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navigation"
>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/browse"
        onClick={[Function]}
      >
        Browse
      </a>
    </li>
  </ul>
</nav>
`;
